### ISSUE TITLE
Downloading results and moving them around may affect the shiny visualizations 

### ISSUE THREAD

**Issue by jfavate** [riboviz/RiboViz#410]
Because the shiny app uses paths from the YAML that the user provides during a riboviz run, it expects to find the results at those locations. If the user is on a headless machine and downloads the results to their local computer (i.e. changes the location of those results), the shiny app may not work, though I believe @3mma-mack has done this and it mostly worked. Solutions include 
1. documenting this 
2. changing the way the shiny app works to instead of reading the yaml, having the user specify the location of the output directory. 

I can see that 2 _could_ work but can already think of one thing that would break, which is the gene specific coverage plots where it needs the dataset argument from the yaml because that ends up as part of the path in the h5 files.

I can try downloading results and moving them around to see if I can break the shiny app and try to make it more robust to this sort of thing.

**Comment by john-s-f** [comment]
I think this behavior will be too fragile prevent issues from happening. If the user runs riboviz on a headless machine and downloads the results to a local machine, it can work but it's easy to make it not work. Issues involving differences among operating systems and paths could cause issues. Foremost though, you would need a copy of R and the related Shiny, tidyverse, H5 (not sure how H5 works on windows), etc, packages on your local system. Presumably, if you were running riboviz elsewhere and wanted to download your results, these things wouldn't be installed on your computer otherwise you'd have run riboviz there. In effect, the user would have to download their results, then download the necessary packages in order to run the shiny server.

**Comment by kavousan** [comment]
@john-s-f How far does this observation go? Should we hold releasing (and putting on the paper) the visualisations? Sorry if barking up the wrong tree.

**Comment by john-s-f** [comment]
No, this isn't worth putting a holdup on release. I think the only way to concretely solve this would be to give users the ability to upload their results to a remote server and have the visualization occur there. This would negate any issues they might have installing software and such and still allow them to download and keep their results locally.

**Comment by FlicAnderson** [comment]
I agree with @john-s-f , perhaps for now we could clearly advise in the documentation that the interactive visualization option works most stably on a non-headless system, and describe why.

**Comment by john-s-f** [comment]
Well it will work locally or on a headless machine, it's just that you shouldn't try to go from one to the other.

**Comment by kavousan** [comment]
@john-s-f are you happy to write the documentation for this?

**Comment by john-s-f** [comment]
Yeah I can add that to the documentation about the shiny app.

**Comment by kavousan** [comment]
Thank John, I created a new ticket, so this ticket here can be about the actual fix. How soon do we need this fixed?
